subprojects/dependency-management/src/test/groovy/org/gradle/internal/component/external/model/DefaultMavenModuleResolveMetadataTest.groovy

```diff
@@ -159,34 +160,26 @@ class DefaultMavenModuleResolveMetadataTest extends AbstractLazyModuleComponentR
-            assert variantsForGraphTraversal[2].attributes.getAttribute(stringUsageAttribute) == "java-api"
-            assert variantsForGraphTraversal[2].attributes.getAttribute(componentTypeAttribute) == "platform"
-            assert variantsForGraphTraversal[3].name == "platform-runtime"
-            assert variantsForGraphTraversal[3].attributes.getAttribute(stringUsageAttribute) == "java-runtime"
-            assert variantsForGraphTraversal[3].attributes.getAttribute(componentTypeAttribute) == "platform"
-            assert variantsForGraphTraversal[4].name == "enforced-platform-compile"
-            assert variantsForGraphTraversal[4].attributes.getAttribute(stringUsageAttribute) == "java-api"
-            assert variantsForGraphTraversal[4].attributes.getAttribute(componentTypeAttribute) == "enforced-platform"
-            assert variantsForGraphTraversal[5].name == "enforced-platform-runtime"
-            assert variantsForGraphTraversal[5].attributes.getAttribute(stringUsageAttribute) == "java-runtime"
-            assert variantsForGraphTraversal[5].attributes.getAttribute(componentTypeAttribute) == "enforced-platform"
-        } else {
-            assert !variantsForGraphTraversal
-        }
+        variantsForGraphTraversal.size() == 6
+        variantsForGraphTraversal[0].name == "compile"
+        variantsForGraphTraversal[0].attributes.getAttribute(stringUsageAttribute) == "java-api"
+        variantsForGraphTraversal[1].name == "runtime"
+        variantsForGraphTraversal[1].attributes.getAttribute(stringUsageAttribute) == "java-runtime"
+        variantsForGraphTraversal[2].name == "platform-compile"
+        variantsForGraphTraversal[2].attributes.getAttribute(stringUsageAttribute) == "java-api"
+        variantsForGraphTraversal[2].attributes.getAttribute(componentTypeAttribute) == "platform"
+        variantsForGraphTraversal[3].name == "platform-runtime"
+        variantsForGraphTraversal[3].attributes.getAttribute(stringUsageAttribute) == "java-runtime"
+        variantsForGraphTraversal[3].attributes.getAttribute(componentTypeAttribute) == "platform"
+        variantsForGraphTraversal[4].name == "enforced-platform-compile"
+        variantsForGraphTraversal[4].attributes.getAttribute(stringUsageAttribute) == "java-api"
+        variantsForGraphTraversal[4].attributes.getAttribute(componentTypeAttribute) == "enforced-platform"
+        variantsForGraphTraversal[5].name == "enforced-platform-runtime"
+        variantsForGraphTraversal[5].attributes.getAttribute(stringUsageAttribute) == "java-runtime"
+        variantsForGraphTraversal[5].attributes.getAttribute(componentTypeAttribute) == "enforced-platform"
 
         where:
-        packaging      | isJavaLibrary
-        "pom"          | true
-        "jar"          | true
-        "maven-plugin" | true
-        "war"          | false
+        packaging << ["pom", "jar", "maven-plugin", "war"]
```

Comment:
Maybe add `aar` in there, since that's one of the known issue with the previous state of the code.

Reply:
added